File: HW13/src/main/java/ru/otus/web/service/GenreService.java
package ru.otus.web.service;

import org.springframework.security.access.annotation.Secured;
import ru.otus.domain.Book;
import ru.otus.domain.Genre;
import ru.otus.web.request.UpdateGenreRequest;

import java.util.List;
import java.util.Optional;

public interface GenreService {

	List<Genre> getAllGenres();

	Optional<Genre> getGenre(String name);

	List<Book> getBooksOfGenre(Genre genre);

	@Secured("ROLE_ADMIN")
	Genre updateGenre(Genre genre, UpdateGenreRequest request);
}

File: final-project/ui/src/services/GenreService.js
// @flow
import RestClient from './RestClient';

import type { Genre, GenreFormValues } from 'types';

const BASE_PATH: string = '/book/genre';

class GenreService {

	static fetchAllGenres(): Promise<Array<Genre>> {
		return RestClient.get(BASE_PATH);
	}

	static updateGenre(name: string, values: GenreFormValues) {
		return RestClient.put(`${BASE_PATH}/${name}`, values);
	}
}

export default GenreService;
